Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,84 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private readonly Stream Source;
+        private bool StreamDisposed = false;
         public DataLoader(Stream source)
         {
+            if(!(source is null))
+            {
+                Source = source;
+            }
+            else
+            {
+                throw new ArgumentNullException("DataLoader's Stream cannot be null");
```

Comment:
This constructor takes in the parameter name, not a message. 
```suggestion
                throw new ArgumentNullException(nameof(source));
```